### PR TITLE
Add tests to wasmate.py

### PR DESCRIPTION
**Pull request by binji**
I copied and modified the test runner from the spec repo.

### CODE REVIEW

**Comment by jfbastien** [comment]
I was actually thinking about using something like the code in #7 as out test suite: import everything from LLVM, and check in the "golden" .wast files. This should keep everything working since the LLVM tests should be comprehensive.

We may want to add tests like this later, but I'm hoping not (LLVM's tests should exercise everything!).

WDYT?

**Comment by binji** [comment]
OK, makes sense. Is there any nice way to get those tests without pulling all of LLVM? I guess via svn?

**Comment by binji** [comment]
OK, svn co ... seems to work OK. But now we need to build LLVM to run these tests. :-(

I guess the alternative is to check in the output from clang. What do you think?

**Comment by jfbastien** [comment]
Checking in the output would work. @flagxor was talking about automating the LLVM build, packaging it and putting it on a cloud bucket. Maybe that output could also be packaged and archived?

**Comment by binji** [comment]
Yeah, that'd be good. I'll check in the output for now, with copied scripts from wasm-e2e to build llvm. If we add the cloud bucket later, we can switch to that instead.

**Comment by flagxor** [comment]
I had some conversations about getting the resources for this last week. I hit some snags, but will try to get something going next week. In the meantime, what you're doing sounds fine.

**Comment by jfbastien** [comment]
That sounds great! lgtm when you check in the generated output.

**Comment by binji** [comment]
PTAL. As you mentioned, some of the tests fail. For now, I'm just leaving them failing.

**Comment by jfbastien** [comment]
Looks good at a quick glance, but I haven't read the code in-depth as I'm about to pop out of existence for 2 weeks :-)

OK to commit for me, since it's a strict improvement over the previous state of things! Maybe we could mark some tests as broken for now, so Travis works?

**Comment by binji** [comment]
Yeah, I'll do that in another CL. Travis isn't even hooked up yet, so there's no worry about breaking anything. :-)

File: prototype-wasmate/scripts/llvm-to-wack.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ROOT_DIR= os.path.dirname(SCRIPT_DIR)
+LLVM_TEST_DIR = os.path.join(ROOT_DIR, 'third_party', 'llvm', 'test', 'CodeGen',
+                             'WebAssembly')
+WACK_TEST_DIR = os.path.join(ROOT_DIR, 'test')
+LLVM_DIR = os.path.join(ROOT_DIR, 'third_party', 'llvm')
+BIN_DIR = os.path.join(LLVM_DIR, 'build', 'bin')
+
+
+def FindTestFiles(directory, ext):
+    tests = []
```
Comment (flagxor) [inline prototype-wasmate/scripts/llvm-to-wack.py]: Your tabbing style is different here. Why?

**Comment by flagxor** [comment]
lgtm with a couple comments

File: prototype-wasmate/run-tests.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+def generate_test_cases(cls, files):
+  for fileName in files:
+    attrName = fileName
+    testCase = generate_test_case(attrName, fileName)
+    setattr(cls, attrName, testCase)
+
+if __name__ == "__main__":
+  try:
+    os.makedirs("test/output/")
```
Comment (flagxor) [inline prototype-wasmate/run-tests.py]: Worth making the script fully indepedent of where it's run?

File: prototype-wasmate/scripts/llvm-to-wack.py (hunk 1 above)
Comment (binji) [inline prototype-wasmate/scripts/llvm-to-wack.py]: Trying to be consistent w/ wasmate.py, 4-space indents. But I see that run-tests.py doesn't have them.

**Pull request merged**